### PR TITLE
Support getting the database product version in the jdbc driver

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
This PR adds support for getting the database product version from the connection metadata.

### CODE REVIEW

**Comment by electrum** [comment]
We definitely need this feature (and for the CLI), but there needs to be a better way to do this. Perhaps return it as an HTTP header?

**Comment by nezihyigitbasi** [comment]
@electrum instead of running a query, this version does a ``HEAD`` request to the ``NodeResource``.

**Comment by electrum** [comment]
We should add a new resource that returns a structured "server info" response. Maybe `/v1/version` or `/v1/info`. Will let @martint decide.

**Comment by martint** [comment]
`/v1/info` sounds good to me

**Comment by nezihyigitbasi** [comment]
@electrum initial version is ready for review.

File: presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java [hunk 1]
```diff
@@ -116,8 +119,13 @@ public String getDatabaseProductName()
     public String getDatabaseProductVersion()
             throws SQLException
     {
-        // TODO: get version from server
-        return "UNKNOWN";
+        Optional<ServerInfo> serverInfo = connection.getServerInfo();
```
Comment (electrum) [inline presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java]: `Optional` plus lambdas are awesome:
```java
return connection.getServerInfo()
        .map(version -> version.getNodeVersion().getVersion())
        .orElse("UNKNOWN");

File: presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java [hunk 2]
```diff
@@ -572,11 +574,22 @@ String getUser()
 
     StatementClient startQuery(String sql)
     {
+        return queryExecutor.startQuery(createClientSession(), sql);
+    }
+
+    Optional<ServerInfo> getServerInfo()
+    {
+        ServerInfo serverInfo = queryExecutor.getServerInfo(createClientSession());
+        return serverInfo != null ? Optional.of(serverInfo) : Optional.empty();
```
Comment (electrum) [inline presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java]: You can use `Optional.ofNullable()`:
```java
return Optional.ofNullable(queryExecutor.getServerInfo(createClientSession()));

**Comment by electrum** [comment]
I'm merging this, the comments are just FYI

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**